src/main/java/com/prajina/academy/controller/ModuleRestController.java

```diff
@@ -56,16 +56,16 @@
 	public ResponseEntity<?> createModule(@RequestBody Module module, UriComponentsBuilder ucBuilder) {
 		logger.debug("Creating Module " + module.getName());
 
-		if (moduleService.isModuleExist(module)) {
+		if (module.getId() != null && moduleService.isModuleExist(module)) {
 			logger.debug("A Module with name " + module.getName() + " already exist");
 			return new ResponseEntity<Void>(HttpStatus.CONFLICT);
```

Comment:
avoid Void return, send back a useful validation message (ideally a validation object)  in the body.